File: 0513-find-bottom-left-tree-value/0513-find-bottom-left-tree-value.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int findBottomLeftValue(TreeNode* root) {
        int level=0;
        queue<TreeNode*>q;
        q.push(root);
         int leftnode;
        while(!q.empty()){
            int size=q.size();
         leftnode= q.front()->val;
            while(size--){
                TreeNode* node= q.front();
                q.pop();
if(node->left) q.push(node->left);
if(node->right) q.push(node->right);
            }
        }
        return leftnode;
    }
};

File: 0513-find-bottom-left-tree-value/0513-find-bottom-left-tree-value.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */
var findBottomLeftValue = function(root) {
    
        const q=[];
        q.push(root);
     let leftnode;
        while(q.length>0){
          let size=q.length;
         leftnode= q[0].val; //don't use shift here
            while(size--){
                const node= q.shift();
                
if(node.left) q.push(node.left);
if(node.right) q.push(node.right);
            }
        }
        return leftnode;
};